### PR TITLE
TASK-39800: Add the "Vote in a date poll" rule and allow voting on all event

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**

### CODE REVIEW

File: agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/date-poll/AgendaEventDatePollDetails.vue [hunk 1]
```diff
@@ -410,12 +410,19 @@ export default {
...
       for(const index in this.dateOptions) {
         const dateOptionId = this.dateOptions[index].id;
         if (this.currentUserVotes.dateOptionVotes[index]) {
-          promises.push(this.$eventService.voteEventDate(eventId, dateOptionId));
+          dateList.push(dateOptionId);
+        }
+      }
+      for(const index in this.dateOptions) {
+        if (this.currentUserVotes.dateOptionVotes[index]) {
+          promises.push(this.$eventService.saveEventVotes(eventId, dateList));
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/date-poll/AgendaEventDatePollDetails.vue]: no neeed to `promises`, you will need to do a single request to server with the list of accepted dateOptions only

File: agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/date-poll/AgendaEventDatePollDetails.vue [hunk 2]
```diff
@@ -410,12 +410,19 @@ export default {
     sendVotes() {
       const eventId = this.event.id;
       const promises = [];
+      const dateList = [];
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/date-poll/AgendaEventDatePollDetails.vue]:
```suggestion
      const acceptedDateOptionIds = [];
```

File: agenda-webapps/src/main/webapp/vue-app/agenda-common/js/EventService.js [hunk 3]
```diff
@@ -280,8 +280,28 @@ export function voteEventDate(eventId, dateOptionId) {
     });
 }
 
+export function saveEventVotes(eventId, dateOptionIds) {
+  const formData = new FormData();
+  Object.keys(dateOptionIds).forEach(dateOptionId => {
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda-common/js/EventService.js]:
```suggestion
  if (dateOptionIds) {
    dateOptionIds.forEach(dateOptionId => {
```

File: agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/date-poll/AgendaEventDatePollDetails.vue (hunk 1 above)
Comment (hassenelkamel-exo) [reply agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/date-poll/AgendaEventDatePollDetails.vue]: for the dismiss we send only the dateOptionId so we need promises

File: agenda-services/src/main/java/org/exoplatform/agenda/listener/AgendaEventGamificationIntegrationListener.java [hunk 4]
```diff
@@ -72,6 +75,18 @@ public void onEvent(Event<Long, Long> event) throws Exception {
           } catch (Exception e) {
             LOG.error("Cannot broadcast gamification event");
           }
+        } else if(StringUtils.equals(eventName, Utils.POST_VOTES_AGENDA_EVENT_POLL)) {
```
Comment (boubaker) [inline agenda-services/src/main/java/org/exoplatform/agenda/listener/AgendaEventGamificationIntegrationListener.java]: code formatting

**Review by boubaker — APPROVED** [review]

**Pull request merged**